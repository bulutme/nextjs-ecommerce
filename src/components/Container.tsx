"use client";
import { ReactNode } from "react";
import styled from "styled-components";

type Props = {
  children: ReactNode;
};

const StyledContainer = styled.div`
  margin-right: auto;
  margin-left: auto;
  padding-left: 0.625rem;
  padding-right: 0.625rem;

  @media (min-width: ${({ theme }) => theme.screens.md}) {
    width: 750px;
  }

  @media (min-width: ${({ theme }) => theme.screens.lg}) {
    width: 960px;
  }

  @media (min-width: ${({ theme }) => theme.screens.xl}) {
    width: 1100px;
  }

  @media (min-width: ${({ theme }) => theme.screens["2xl"]}) {
    width: 1350px;
  }
`;

const Container = ({ children }: Props) => {
  return <StyledContainer>{children}</StyledContainer>;
};

export default Container;
