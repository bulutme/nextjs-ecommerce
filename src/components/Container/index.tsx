import { FC, ReactNode } from "react";
import styled from "styled-components";

type Props = {
  children: ReactNode;
};

const StyledContainer = styled.div`
  margin-right: auto;
  margin-left: auto;
  padding-left: 0.625rem;
  padding-right: 0.625rem;

  width: auto;

  @media (max-width: ${({ theme }) => theme.screens["2xs"]}) {
    width: 350px;
  }

  @media (min-width: ${({ theme }) => theme.screens.md}) {
    width: 750px;
  }

  @media (min-width: ${({ theme }) => theme.screens.lg}) {
    width: 960px;
  }

  @media (min-width: ${({ theme }) => theme.screens.xl}) {
    width: 1100px;
  }

  @media (min-width: ${({ theme }) => theme.screens["2xl"]}) {
    width: 1350px;
  }
`;

const Container: FC<Props> = ({ children }) => {
  return <StyledContainer>{children}</StyledContainer>;
};

export default Container;
