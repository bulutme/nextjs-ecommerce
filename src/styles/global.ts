"use client";

import { createGlobalStyle } from "styled-components";

const GlobalStyles = createGlobalStyle`
  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }

  html {
    font-size: 16px;
  }

  html, body, #__next {
    height: 100%;
  }

  a {
    text-decoration: none;
    width: fit-content;
  }

  ul {
    list-style-type: none
  }
`;

export default GlobalStyles;
