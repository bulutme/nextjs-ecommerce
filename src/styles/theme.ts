export const theme = {
  colors: {
    black: "#000",
    white: "#fff",
    gray: "#808080",
    grayLight: "#d3d3d3",
    border: "#e2e8f0",
    primary: "#3bb77e",
    primaryDark: "#2b8a5e",
    error: "#dc2626",
    info: "#0891b2",
    success: "#16a34a",
  },
  screens: {
    "2xs": "400px",
    xs: "480px",
    sm: "576px",
    md: "768px",
    lg: "992px",
    xl: "1200px",
    "2xl": "1440px",
    "3xl": "1560px",
  },
  fontWeight: {
    normal: 400,
    medium: 500,
    semibold: 600,
    bold: 700,
  },
};
