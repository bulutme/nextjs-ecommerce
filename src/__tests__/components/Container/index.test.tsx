import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import Container from "@/components/Container";
import { theme } from "@/styles/theme";

const renderWithTheme = (component: React.ReactElement) =>
  render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);

describe("Container Component", () => {
  test("renders correctly", () => {
    const { container } = renderWithTheme(
      <Container>
        <div>Child Content</div>
      </Container>
    );

    expect(container).toMatchSnapshot();
  });
});
