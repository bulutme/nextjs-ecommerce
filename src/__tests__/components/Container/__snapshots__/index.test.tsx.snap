// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Component renders correctly 1`] = `
<div>
  <div
    class="sc-f2eef38d-0 iwNFqL"
  >
    <div>
      Child Content
    </div>
  </div>
</div>
`;
